Expensify/react-native-live-markdown
# Fix Live Markdown Input undo/redo history on web

### Details

This PR enhances undo/redo history logic. The main changes are:
- changed the `debouncedAdd` function to `throttledAdd`. Now it instantly adds an item to history and until the throttle time ends, executing it again will only modify the previously added item. After the time ends, launching `throttledAdd` will add a new element.
Before these changes `debouncedAdd` function was adding an element to history after the debounce time 
- fixed cursor positioning when undoing text. Now cursor is positioned at the beginning of the text that was undone

### Related Issues

https://github.com/Expensify/App/issues/39255

### Manual Tests

1. Copy and paste text into Live Markdown Input somewhere else than the end of the value inside
2. Try to undo it immediately after pasting it
3. Verify if text was undone and if cursor position was set it the place where the undo text was beginning

### Linked PRs
